src/Files.App/Views/Properties/HashesPage.xaml

```diff
@@ -25,21 +26,79 @@
 	</vm:BasePropertiesPage.Resources>
 
 	<Grid x:Name="RootGrid">
+		<Grid.RowDefinitions>
+			<RowDefinition Height="Auto" />
+			<RowDefinition Height="Auto" />
+			<RowDefinition Height="*" />
+		</Grid.RowDefinitions>
 
+		<!--  Hash Comparison Section  -->
 		<Grid
-			x:Name="HashesListGrid"
-			Margin="12"
+			x:Name="HashComparisonGrid"
+			Margin="12,12,12,4"
 			VerticalAlignment="Top"
 			Background="{ThemeResource CardBackgroundFillColorDefaultBrush}"
 			BorderBrush="{ThemeResource CardStrokeColorDefaultBrush}"
 			BorderThickness="1"
 			CornerRadius="4">
+			<Grid.ColumnDefinitions>
+				<ColumnDefinition Width="*" />
+				<ColumnDefinition Width="Auto" />
+			</Grid.ColumnDefinitions>
+
+			<!--  Hash Input  -->
+			<Grid Grid.Column="0" Margin="12">
+				<Grid.ColumnDefinitions>
+					<ColumnDefinition Width="*" />
+				</Grid.ColumnDefinitions>
+
+				<TextBox
+					x:Name="HashInputTextBox"
+					Grid.Column="0"
+					PlaceholderText="{helpers:ResourceString Name=EnterHashToCompare}"
+					Text="{x:Bind HashesViewModel.HashInput, Mode=TwoWay, UpdateSourceTrigger=PropertyChanged}">
+					<interactivity:Interaction.Behaviors>
+						<interactivity:EventTriggerBehavior EventName="TextChanged">
+							<interactivity:InvokeCommandAction Command="{x:Bind HashesViewModel.HashInputTextChangedCommand}" />
+						</interactivity:EventTriggerBehavior>
+					</interactivity:Interaction.Behaviors>
+				</TextBox>
+			</Grid>
+
+			<!--  Compare File Button  -->
+			<Button
+				x:Name="CompareFileButton"
+				Grid.Column="1"
+				Margin="12"
+				Command="{x:Bind HashesViewModel.CompareFileCommand}"
+				Content="{helpers:ResourceString Name=CompareFile}" />
+		</Grid>
+
+		<InfoBar
+			x:Name="HashMatchInfoBar"
+			Grid.Row="1"
+			Margin="12,4,12,4"
+			x:Load="{x:Bind HashesViewModel.IsInfoBarOpen, Mode=OneWay}"
+			IsClosable="False"
+			IsOpen="True"
+			Message="{x:Bind HashesViewModel.InfoBarTitle, Mode=OneWay}"
+			Severity="{x:Bind HashesViewModel.InfoBarSeverity, Mode=OneWay}" />
+
+		<Grid
+			x:Name="HashesListGrid"
+			Grid.Row="2"
+			Margin="12,4,12,12"
+			VerticalAlignment="Stretch"
+			Background="{ThemeResource CardBackgroundFillColorDefaultBrush}"
+			BorderBrush="{ThemeResource CardStrokeColorDefaultBrush}"
+			BorderThickness="1"
+			CornerRadius="4">
 
 			<!--  Hashes ListView  -->
 			<ListView
 				x:Name="HashesListView"
-				ItemsSource="{x:Bind HashesViewModel.Hashes, Mode=TwoWay}"
-				SelectedItem="{x:Bind HashesViewModel.SelectedItem, Mode=TwoWay}">
+				ItemsSource="{x:Bind HashesViewModel.Hashes, Mode=OneWay}"
+				SelectedItem="{x:Bind HashesViewModel.SelectedItem, Mode=OneWay}">
```

Comment:
SelectedItem might need to be `TwoWay`.

Reply:
I'll just revert both so they are what they were before.

Reply:
We might as well fix `ItemsSource`, there is no reason for it to be `TwoWay`.